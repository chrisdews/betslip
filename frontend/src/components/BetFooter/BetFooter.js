import React from "react";
import PropTypes from "prop-types";
import Button from "../Button";
import styled from "styled-components";
function BetFooter({ betButtonDisabled, setBetPlaced }) {
  const placeBetClickHandler = () => {
    setBetPlaced(true);
  };
  return (
    <FooterStyle>
      <BetButton>
        <Button
          clickHandler={placeBetClickHandler}
          buttonDisabled={betButtonDisabled}
        >
          BET NOW
        </Button>
      </BetButton>
    </FooterStyle>
  );
}

BetFooter.propTypes = {
  betButtonDisabled: PropTypes.bool,
  setBetPlaced: PropTypes.func,
};

export default BetFooter;

const FooterStyle = styled.div`
  background-color: white;
  display: flex;
  justify-content: space-between;
  align-items: right;
  padding: 0em 1em 0em 1em;
`;

const BetButton = styled.span`
  margin-left: auto;
  margin-bottom: 1em;
`;
