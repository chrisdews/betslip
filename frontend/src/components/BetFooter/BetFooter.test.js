import React from "react";
import BetFooter from "./BetFooter";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("BetFooter", () => {
  it("renders correctly - snapshot", () => {
    const { container } = render(<BetFooter />);
    expect(container).toMatchSnapshot();
  });

  it("the BET NOW button is disabled when passed betButtonDisabled is true", () => {
    const { getByText } = render(<BetFooter betButtonDisabled={true} />);
    expect(getByText(/bet now/i)).toBeDisabled();
  });

  it("the BET NOW button is disabled when passed betButtonDisabled is false", () => {
    const { getByText } = render(<BetFooter betButtonDisabled={false} />);
    expect(getByText(/bet now/i)).not.toBeDisabled();
  });

  it("BET NOW button on click calls setBetPlaced with an argument of true", () => {
    const testFunction = jest.fn();
    const { getByText } = render(
      <BetFooter betButtonDisabled={false} setBetPlaced={testFunction} />
    );
    fireEvent.click(getByText(/bet now/i));
    expect(testFunction).toBeCalledWith(true);
  });
});
