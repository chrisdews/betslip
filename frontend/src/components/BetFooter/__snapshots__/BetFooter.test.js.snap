// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetFooter renders correctly - snapshot 1`] = `
<div>
  <div
    class="sc-gtsrHT gvNYYZ"
  >
    <span
      class="sc-dlnjwi deoBsm"
    >
      <button
        class="sc-bdnxRM hCETbq"
      >
        BET NOW
      </button>
    </span>
  </div>
</div>
`;
